common/common.go

```diff
@@ -433,28 +434,74 @@ func InArray(val, array interface{}) (exists bool, index int) {
 	return
 }
 
-// Errors defines multiple errors
-type Errors []error
+// AppendError appends error in a more idiomatic way. This can start out as a
+// standard error e.g. err := errors.New("random error")
+// err = AppendError(err, errors.New("another random error"))
+func AppendError(original, incoming error) error {
+	if incoming == nil {
+		if _, file, no, ok := runtime.Caller(1); ok {
+			log.Errorf(log.Global, "AppendError() failed: incoming error is nil %s#%d\n", file, no)
+		}
+		if errSlice, ok := original.(*multiError); ok {
+			errSlice.offset = 0
+		}
```

Comment:
Looks like errSlice is declared and you set offset = 0 in both paths. 
You could kinda sorta do the following, 
```
errSliceP, ok := original.(*multiError)
	if ok {
		errSliceP.offset = 0
	}
	if incoming == nil {
		return original // Skip append - continue as normal.
	}
	if !ok {
		// This assumes that a standard error is passed in and we can want to
		// track it and add additional errors.
		errSliceP = &multiError{}
		if original != nil {
			errSliceP.errors = append(errSliceP.errors, original)
		}
	}
	if incomingSlice, ok := incoming.(*multiError); ok {
		// Join slices if needed.
		errSliceP.errors = append(errSliceP.errors, incomingSlice.errors...)
	} else {
		errSliceP.errors = append(errSliceP.errors, incoming)
	}
	return errSliceP
```
But it doesn't flow as nicely. So this is pretty optional